### PR TITLE
Nft Gallery

### PR DESCRIPTION
**Pull request by derHowie**
Figma link (if any): https://www.figma.com/file/VyqXrnmqNUQ4iBQxbwPxka/BX-nfts?type=design&node-id=0-1&mode=design&t=AgRkleTwuL7gMnts-0

## What changed (plus any additional context for devs)
Added content in the NFT tab behind a feature flag.

This ticket covers displaying nfts in gallery and collection mode. This also includes the introduction of nft sort.

There are some notable things not included in this PR:
- there is no special NFT fallback image state, so the alien displays there (https://linear.app/rainbow/issue/BX-1100/nft-image-fallback)
- there is no null/loading state for the NFT tab, so a brand new wallet will show the prerelease UI while loading (https://linear.app/rainbow/issue/BX-1105/empty-state-nft-gallery)
- the copy/design in the displayMode/sort dropdowns has not been finalized 
- keyboard navigation is not fully supported (https://linear.app/rainbow/issue/BX-1095/keyboard-navigation-nft-gallery)
- keyboard shortcuts are not included (https://linear.app/rainbow/issue/BX-1097/shortcuts-nft-gallery)
- I tested this UI with wallets that owned 5k < nfts. I noticed when attempting to open Vitalik's wallet (40k nfts) that my current networking implementation will not suffice. I think we should put off stress testing until this is completed: https://linear.app/rainbow/issue/BX-1098/networking-infinite-nft-query + https://linear.app/rainbow/issue/BX-1107/performance-audit-behavior-of-wallets-with-10k-nfts

env vars: https://github.com/rainbow-me/browser-extension-env/pull/16

## Screen recordings / screenshots
PoW: https://recordit.co/awEdiIbcBR

## What to test
- nfts display
- display modes work
- sorting works
- no crashes
- wallet switching / importing doesn't break NFT UI
- make sure feature flag appropriately hides the new UI

### CODE REVIEW

File: src/core/state/currentSettings/featureFlags.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ export const featureFlagsStore = createStore<FeatureFlagsStore>(
       hw_wallets_enabled: true,
       command_k_internal_shortcuts_enabled: false,
       custom_rpc: false,
+      nfts_enabled: false,
```
Comment (estebanmino) [inline src/core/state/currentSettings/featureFlags.ts:26]: you need to bump the persist version to make this new feature flag available 
i spent some time trying to make it appear

**Review by estebanmino — COMMENTED** [review]
this was probably needed on your previous pr but NFT_API_URL is not available as env var so nfts don't work, am i missing something?

**Review by estebanmino — COMMENTED** [review]
should we have these two sort buttons when we're showing the coming soon text?

image

i'm thinking we're gonna merge this and they're going to be available even tho the feature flag is off

EDIT: is not visible when the flag off, just tried, but if you don't have nfts you'll see the coming soon message with the sort buttons, not sure if intended

**Review by estebanmino — COMMENTED** [review]
these dropdowns should probably have a distance to the dropdown trigger

image

as we have on others

image

File: src/core/state/nfts/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    address: Address;
+    collectionId: string;
+  }) => void;
+}
+
+export const nftsStore = createStore<NftsState>(
+  (set, get) => ({
+    displayMode: 'grouped',
+    sections: {},
+    sort: 'recent',
```
Comment (brunobar79) [inline src/core/state/nfts/index.ts:29]: not a fan of having strings here since this is used in a few places

**Review by brunobar79 — APPROVED** [review]
🤝 🤝 🤝

**Review by estebanmino — APPROVED** [review]
🖼️

**Pull request merged**